src/core_plugins/elasticsearch/lib/create_proxy.js

```diff
@@ -17,61 +17,57 @@
  * under the License.
  */
 
-import createAgent from './create_agent';
-import mapUri from './map_uri';
-import { assign } from 'lodash';
+import Joi from 'joi';
 
-export function createPath(prefix, path) {
-  path = path[0] === '/' ? path : `/${path}`;
-  prefix = prefix[0] === '/' ? prefix : `/${prefix}`;
+export function createProxy(server) {
+  const { callWithRequest } =  server.plugins.elasticsearch.getCluster('data');
 
-  return `${prefix}${path}`;
-}
-
-export function createProxy(server, method, path, config) {
-  const proxies = new Map([
-    ['/elasticsearch', server.plugins.elasticsearch.getCluster('data')],
-  ]);
-
-  const responseHandler = function (err, upstreamResponse, request, reply) {
-    if (err) {
-      reply(err);
-      return;
-    }
+  server.route({
+    method: 'POST',
+    path: '/elasticsearch/_msearch',
+    config: {
+      payload: {
+        parse: 'gunzip'
+      }
+    },
+    async handler(req, reply) {
+      const { payload } = req;
+      try {
+        const body = payload
+          .toString('utf8').split('\n')
+          .filter(Boolean)
+          .map(JSON.parse);
```

Comment:
this behavior's different, previously a parse error would still make it to elasticsearch.